### PR TITLE
Scheduler Grouping: rewrite

### PR DESCRIPTION
**Pull request by pavel-2020**
[Trello](https://trello.com/c/lTA2NdUk/1668-scheduler-demos-grouping-2-demos)
[Demo](https://js.devexpress.com/Demos/WidgetsGallery/Demo/Scheduler/GroupOrientation)
CPs: https://github.com/DevExpress/devextreme-demos/pull/962, https://github.com/DevExpress/devextreme-demos/pull/963

### CODE REVIEW

File: JSDemos/Demos/Scheduler/GroupOrientation/description.md [hunk 1]
```diff
@@ -1 +1,3 @@
-This example shows the **workWeek** view with different group orientations. Select the *"Vertical Grouping"* tab to&nbsp;see the resource headers organized in&nbsp;a&nbsp;column and the *"Horizontal Grouping"*&nbsp;&mdash; arranged in&nbsp;a&nbsp;row.
\ No newline at end of file
+If resources are assigned to appointments, you can group them by these resources. To do so, use the [groups](/api-reference/10%20UI%20Components/dxScheduler/1%20Configuration/groups.md '/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/#groups') property to specify an array. Each element of this array is **fieldExpr** of a resource kind. Note that the resource headers' order depends on the resources' order in the [resources](/api-reference/10%20UI%20Components/dxScheduler/1%20Configuration/resources '/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/') array.
```
Comment (AlexJameson) [inline JSDemos/Demos/Scheduler/GroupOrientation/description.md]: >If resources are assigned to appointments, you can group them by these resources. 

Please add links to appointments and resources.

>To do so, use the [groups](/api-reference/10%20UI%20Components/dxScheduler/1%20Configuration/groups.md '/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/#groups') property to specify an array. 

This sentence breaks suddenly. How about changing it a bit as follows: To do this, specify the [groups[]]() array. 

>Each element of this array is **fieldExpr** of a resource kind. 

Let's add a link.
 
>Note that the resource headers' order depends on the resources' order in the [resources]() array.

Once again, I suggest you not to use apostrophes with plurals. How about this: Note that the order of resource headers depends on the order of elements in the **resources[]** array.

File: JSDemos/Demos/Scheduler/GroupOrientation/description.md [hunk 2]
```diff
@@ -1 +1,3 @@
...
+
+The Scheduler arranges group headers vertically (in a column) or horizontally (in a row). This depends on the view type. You can use the **views**.[groupOrientation](/api-reference/10%20UI%20Components/dxScheduler/1%20Configuration/views/groupOrientation.md '/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/views/#groupOrientation') property to set custom orientation for specific views.
```
Comment (AlexJameson) [inline JSDemos/Demos/Scheduler/GroupOrientation/description.md]:
```suggestion
The Scheduler arranges group headers vertically (in a column) or horizontally (in a row). This depends on the view [type](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/views/#type). You can use the **views**.[groupOrientation](/api-reference/10%20UI%20Components/dxScheduler/1%20Configuration/views/groupOrientation.md '/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/views/#groupOrientation') property to set custom orientation for specific views.
```

**Review by AlexJameson — DISMISSED** [review]

File: JSDemos/menuMeta.json [hunk 3]
```diff
@@ -1788,13 +1788,12 @@
         ]
       },
       {
-        "Name": "Grouping",
+        "Name": "Group Appointments",
```
Comment (RomanTsukanov) [inline JSDemos/menuMeta.json]: We aggreed to keep the old names.

File: JSDemos/Demos/Scheduler/GroupOrientation/description.md [hunk 4]
```diff
@@ -1 +1,3 @@
-This example shows the **workWeek** view with different group orientations. Select the *"Vertical Grouping"* tab to&nbsp;see the resource headers organized in&nbsp;a&nbsp;column and the *"Horizontal Grouping"*&nbsp;&mdash; arranged in&nbsp;a&nbsp;row.
\ No newline at end of file
+If [resources](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/) are assigned to appointments, you can group them by these resources. To do this, specify the [groups](/api-reference/10%20UI%20Components/dxScheduler/1%20Configuration/groups.md '/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/#groups') array. Each element of this array is [fieldExpr](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#fieldExpr) of a resource kind. Note that the order of resource fields depends on the order of elements in the **resources** array.
```
Comment (RomanTsukanov) [inline JSDemos/Demos/Scheduler/GroupOrientation/description.md]: Let's look at the description from the standpoint of a reader who doesn't know about resource. We should first introduce him to the concept, give a link to the Resources demo, and only then go on about grouping.

File: JSDemos/Demos/Scheduler/GroupOrientation/description.md [hunk 5]
```diff
@@ -1 +1,3 @@
...
+
+The Scheduler arranges group headers vertically (in a column) or horizontally (in a row). This depends on the view [type](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/views/#type). You can use the **views**.[groupOrientation](/api-reference/10%20UI%20Components/dxScheduler/1%20Configuration/views/groupOrientation.md '/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/views/#groupOrientation') property to set custom orientation for specific views.
```
Comment (RomanTsukanov) [inline JSDemos/Demos/Scheduler/GroupOrientation/description.md]: Use only URLs in links within demo descriptions.

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

File: JSDemos/Demos/Scheduler/GroupOrientation/description.md [hunk 6]
```diff
@@ -1 +1,5 @@
...
\ No newline at end of file
+[Resources](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/) are categories used to classify appointments. To understand what resources are, consider the following example: in an educational center lectures are held in several rooms; in Scheduler terms, room is a resource kind, individual rooms are resource instances, and lectures are appointments that use these resource instances.
+
+The [Resources demo](/Demos/WidgetsGallery/Demo/Scheduler/Resources/) describes how to assign resources. When assigned, they can also be used to group appointments. To implement this, specify the [groups[]](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/#groups) array. Each element of this array is [fieldExpr](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#fieldExpr) of a resource kind. The order of resource headers in the UI is the same as the order of the resource instances. Note that if the **resources** array has more than one element, grouping is nested.
```
Comment (RomanTsukanov) [inline JSDemos/Demos/Scheduler/GroupOrientation/description.md]:
```suggestion
To group appointments by resources, specify the [groups[]](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/#groups) array. Each element of this array is [fieldExpr](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#fieldExpr) of a resource kind. The order of group headers in the UI mirrors the order of resource instances. If the **groups[]** array contains more than one element, groups are nested. 
```

File: JSDemos/Demos/Scheduler/GroupOrientation/description.md [hunk 7]
```diff
@@ -1 +1,5 @@
-This example shows the **workWeek** view with different group orientations. Select the *"Vertical Grouping"* tab to&nbsp;see the resource headers organized in&nbsp;a&nbsp;column and the *"Horizontal Grouping"*&nbsp;&mdash; arranged in&nbsp;a&nbsp;row.
\ No newline at end of file
+[Resources](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/) are categories used to classify appointments. To understand what resources are, consider the following example: in an educational center lectures are held in several rooms; in Scheduler terms, room is a resource kind, individual rooms are resource instances, and lectures are appointments that use these resource instances.
```
Comment (RomanTsukanov) [inline JSDemos/Demos/Scheduler/GroupOrientation/description.md]: We do need to make a connection between grouping and resources. However, this demo is about grouping, and we shouldn't dive deep into explaining the resources concept. The following description would suffice:

```suggestion
Appointments can be grouped by resources. This demo declares a single resource&mdash;Priority. Refer to the following demo for more information about resources and their configuration: [Resources](/Demos/WidgetsGallery/Demo/Scheduler/Resources/).
```

File: JSDemos/Demos/Scheduler/GroupOrientation/description.md [hunk 8]
```diff
@@ -1 +1,5 @@
...
+
+The [Resources demo](/Demos/WidgetsGallery/Demo/Scheduler/Resources/) describes how to assign resources. When assigned, they can also be used to group appointments. To implement this, specify the [groups[]](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/#groups) array. Each element of this array is [fieldExpr](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#fieldExpr) of a resource kind. The order of resource headers in the UI is the same as the order of the resource instances. Note that if the **resources** array has more than one element, grouping is nested. 
+
+The Scheduler arranges group headers vertically (in a column) or horizontally (in a row). You can use the **views**.[groupOrientation](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/views/#groupOrientation) property to set custom orientation for specific views. This property description includes default view orientation types.
```
Comment (RomanTsukanov) [inline JSDemos/Demos/Scheduler/GroupOrientation/description.md]: > The Scheduler arranges

Without "can", it reads as if the Scheduler chooses group orientation randomly.

> This property description includes default view orientation types.

It goes without saying. We better refer to the demo instead.


```suggestion
The Scheduler can arrange group headers vertically (in a column) or horizontally (in a row). You can use the **views**.[groupOrientation](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/views/#groupOrientation) property to set custom orientation for specific views. This demo shows two Work Week views, each of which has a different group orientation.
```

File: JSDemos/Demos/Scheduler/GroupOrientation/description.md (hunk 6 above)
Comment (RomanTsukanov) [reply JSDemos/Demos/Scheduler/GroupOrientation/description.md]: > The order of resource headers in the UI is the same as the order of the resource instances.

By "resource headers" you mean "group headers"? "the order of the resource instances" in which array?

> Note that if the **resources** array has more than one element, grouping is nested. 

Do you mean "if the **groups** array has more than one element"?

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

**Review by RomanTsukanov — DISMISSED** [review]

File: JSDemos/Demos/Scheduler/GroupOrientation/description.md [hunk 9]
```diff
@@ -1 +1,5 @@
-This example shows the **workWeek** view with different group orientations. Select the *"Vertical Grouping"* tab to&nbsp;see the resource headers organized in&nbsp;a&nbsp;column and the *"Horizontal Grouping"*&nbsp;&mdash; arranged in&nbsp;a&nbsp;row.
\ No newline at end of file
+Appointments can be grouped by resources. This demo declares a single resource&mdash;Priority. Refer to the following demo for more information about resources and their configuration: [Resources](/Demos/WidgetsGallery/Demo/Scheduler/Resources/).
```
Comment (NikolaevaIrina) [inline JSDemos/Demos/Scheduler/GroupOrientation/description.md]:
```suggestion
Users can group appointments by resources. This demo declares a single resource&mdash;Priority. Refer to the following demo for more information about resources and their configuration: [Resources](/Demos/WidgetsGallery/Demo/Scheduler/Resources/).
```

File: JSDemos/Demos/Scheduler/GroupOrientation/description.md [hunk 10]
```diff
@@ -1 +1,5 @@
...
+
+To group appointments by resources, specify the [groups[]](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/#groups) array. Each element of this array is [fieldExpr](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#fieldExpr) of a resource kind. The order of group headers in the UI mirrors the order of resource instances. If the **groups[]** array contains more than one element, groups are nested.
```
Comment (NikolaevaIrina) [inline JSDemos/Demos/Scheduler/GroupOrientation/description.md]:
```suggestion
To group appointments by resources, specify the [groups[]](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/#groups) array. Each element of this array is the [fieldExpr](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#fieldExpr) of a corresponding resource kind. The order of group headers in the UI mirrors the order of resource instances in the array. If the **groups[]** array contains more than one element, groups are nested. 
```

File: JSDemos/Demos/Scheduler/GroupOrientation/description.md [hunk 11]
```diff
@@ -1 +1,5 @@
...
\ No newline at end of file
+Appointments can be grouped by resources. This demo declares a single resource&mdash;Priority. Refer to the following demo for more information about resources and their configuration: [Resources](/Demos/WidgetsGallery/Demo/Scheduler/Resources/).
+
+To group appointments by resources, specify the [groups[]](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/#groups) array. Each element of this array is [fieldExpr](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#fieldExpr) of a resource kind. The order of group headers in the UI mirrors the order of resource instances. If the **groups[]** array contains more than one element, groups are nested. 
+
+The Scheduler can arrange group headers vertically (in a column) or horizontally (in a row). You can use the **views**.[groupOrientation](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/views/#groupOrientation) property to set custom orientation for specific views. This demo shows two Work Week views, each of which has a different group orientation.
```
Comment (NikolaevaIrina) [inline JSDemos/Demos/Scheduler/GroupOrientation/description.md]:
```suggestion
The Scheduler can arrange group headers vertically (in a column) or horizontally (in a row). Use the **views**.[groupOrientation](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/views/#groupOrientation) property to set a custom orientation for specific views. This demo shows two Work Week views with different group orientations.
```

File: JSDemos/Demos/Scheduler/GroupOrientation/description.md [hunk 12]
```diff
@@ -1 +1,5 @@
-This example shows the **workWeek** view with different group orientations. Select the *"Vertical Grouping"* tab to&nbsp;see the resource headers organized in&nbsp;a&nbsp;column and the *"Horizontal Grouping"*&nbsp;&mdash; arranged in&nbsp;a&nbsp;row.
\ No newline at end of file
+Users can group appointments by resources. This demo declares a single resource&mdash;Priority. Refer to the following demo for more information about resources and their configuration: [Resources](/Demos/WidgetsGallery/Demo/Scheduler/Resources/).
+
+To group appointments by resources, specify the [groups[]](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/#groups) array. Each element of this array is the [fieldExpr](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#fieldExpr) of a corresponding resource kind. The order of group headers in the UI mirrors the order of items in the resource instances array. If the **groups[]** array contains more than one element, groups are nested.
```
Comment (dirk-pieterse) [inline JSDemos/Demos/Scheduler/GroupOrientation/description.md]:
```suggestion
To group appointments by resources, specify the [groups[]](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/#groups) array. Each element of this array is the [fieldExpr](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/resources/#fieldExpr) of a corresponding resource kind. The order of group headers in the UI is the same as items in the resource instances array. If the **groups[]** array contains more than one element, groups are nested. 
```

**Review by dirk-pieterse — DISMISSED** [review]

**Review by RomanTsukanov — APPROVED** [review]

**Pull request merged**